stools/clair/__init__.py

```diff
@@ -17,75 +17,125 @@
 You should have received a copy of the GNU Affero General Public License
 along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-'''
+"""
 
-from stools.version import __version__
-from stools.clair.image import export_to_targz
-from stools.clair.api import Clair
-from stools.clair.server import start
-from multiprocessing import Process
 import argparse
-import os
-import requests
+import json
 import shutil
 import sys
-import tempfile
-import time
+from multiprocessing import Process
+from os import rename
+from os.path import isdir, join, basename, splitext
+from time import sleep
 
+import requests
+from json2html import *
+from lxml import etree, html
 
-def get_parser():
-    parser = argparse.ArgumentParser(description="Singularity Clair Scanner")
+from stools.clair.api import Clair
+from stools.clair.image import export_to_targz
+from stools.clair.server import start
+from stools.version import __version__
 
-    parser.add_argument('--version', dest="version", 
-                        help="show version and exit.", 
-                        default=False, action='store_true')
 
-    parser.add_argument('--start-server', dest="server", 
-                        help="If running natively, start the web server too.", 
-                        default=True, action='store_true')
+def get_parser():
+    parser = argparse.ArgumentParser(prog="Singularity Clair Scanner", usage=usage())
+
+    parser.add_argument(
+        "--version",
+        action="version",
+        version=version(),
+        dest="version",
+        help="show version and exit",
+        default=False,
+    )
+
+    parser.add_argument(
+        "--start-server",
+        dest="server",
+        help="if running natively, start the web server too",
+        default=True,
+        action="store_true",
+    )
+
+    parser.add_argument(
+        "images", nargs="*", help="Singularity images to scan", type=str
+    )
+
+    parser.add_argument(
+        "--port",
+        default=8080,
+        help="port to serve application (default 8080)",
+        type=int,
+    )
+
+    parser.add_argument(
+        "--report",
+        nargs="?",
+        const="/code/reports",
+        dest="report",
+        help="save Clair reports to chosen directory (default: /code/reports)",
+        type=dir_path,
+    )
+
+    parser.add_argument(
+        "--report-format",
+        choices=["json", "html"],
+        default="json",
+        dest="report_format",
+        help="reports will be saved in specified format when --report passed (default: json)",
+        type=str,
+    )
```

Comment:
Generally you don’t want to add linting to a repo before discussing it first.

Reply:
Understood

Reply:
Reverted :+1: